dspace-api/src/test/java/org/dspace/core/ContextTest.java

```diff
@@ -255,6 +261,60 @@ public void testComplete2() throws SQLException {
         cleanupContext(instance);
     }
 
+    /**
+     * Test of commit method, of class Context.
+     */
+    @Test
+    public void testCommit() throws SQLException, AuthorizeException, IOException {
+        // To test commit() we need a new Context object
+        Context instance = new Context();
+
+        // By default, we should have a new DB connection, so let's make sure it is there
+        assertThat("HibernateDBConnection should exist", instance.getDBConnection(), notNullValue());
+        assertTrue("Context should be valid", instance.isValid());
+        assertTrue("Transaction should be open", instance.isTransactionAlive());
+
+        // Allow full Admin perms (in new context)
+        when(authorizeServiceSpy.isAdmin(instance)).thenReturn(true);
+
+        // Create a new EPerson (to be committed)
+        String createdEmail = "myfakeemail@gmail.com";
+        EPerson newUser = ePersonService.create(instance);
+        newUser.setFirstName(instance, "Tim");
+        newUser.setLastName(instance, "Smith");
+        newUser.setEmail(createdEmail);
+        newUser.setCanLogIn(true);
+        newUser.setLanguage(instance, I18nUtil.getDefaultLocale().getLanguage());
+
+        // Now, call commit()
+        instance.commit();
+
+        // We expect our DB connection to still exist
+        assertThat("HibernateDBConnection should still be open", instance.getDBConnection(), notNullValue());
+        // We expect the Context to be valid
+        assertTrue("Context should still be valid", instance.isValid());
```

Comment:
NOTE for reviewers: This is the test that will fail on current `master`.  Anytime `context.commit()` is called, `context.isValid()` will return `false` until a new database transaction is started in that same Context.